docs/tutorials/tiled_integration/tiled_server.py

```diff
@@ -0,0 +1,54 @@
+import contextlib
+import threading
+import time
+import uvicorn
+import tempfile
+import pathlib
+
+from tiled.server.app import build_app
+from tiled.catalog import from_uri as catalog_from_uri
+
+
+class ThreadedServer(uvicorn.Server):
+
+    def install_signal_handlers(self):
+        pass
+
+    @contextlib.contextmanager
+    def run_in_thread(self):
+        thread = threading.Thread(target=self.run)
+        thread.start()
+        try:
+            # Wait for server to start up, or raise TimeoutError.
+            for _ in range(100):
+                time.sleep(0.1)
+                if self.started:
+                    break
+            else:
+                raise TimeoutError("Server did not start in 10 seconds.")
+            host, port = self.servers[0].sockets[0].getsockname()
+            yield f"http://{host}:{port}"
+        finally:
+            self.should_exit = True
+            thread.join()
+
+
+class TiledServer:
+    
+    def __init__(self, port=8000, dir_path=None, api_key="secret"):
```

Comment:
If the default port is `0`, the kernel will provide a random unused port, eliminating the failure mode "The users was already running Tiled or Jupyter or whatever on that port and now has an error message." Especially useful if the user might be running multiple notebooks and thus have a collision if we hard-code a default.

Reply:
Yep, thanks for this! I wasn't sure is `port=0` should be default, or it should error if 8000 is already taken and let the user set `port=0` themselves. I think it makes sense to set it to a random port by default. Fixed!